### PR TITLE
launcher watchdog wip

### PR DESCRIPTION
**Pull request by luc1an24**

### CODE REVIEW

File: src/NiceHashMinerLauncher/App.xaml.cs [hunk 1]
```diff
@@ -273,8 +273,12 @@ private async void App_OnStartup(object sender, StartupEventArgs e)
                     WindowStyle = ProcessWindowStyle.Normal
                 };
                 var run = true;
+                const int maxRestartCount = 3;
```
Comment (nathsmaR) [inline src/NiceHashMinerLauncher/App.xaml.cs]: 3N4FAhGGXn8Q39kWhBQeNKQcEfM2NiqP4C

File: src/NiceHashMinerLauncher/App.xaml.cs [hunk 2]
```diff
@@ -284,66 +288,112 @@ private async void App_OnStartup(object sender, StartupEventArgs e)
                             niceHashMiner?.WaitForExit();
                             // TODO 
                             Console.WriteLine(niceHashMiner.ExitCode);
-                            // if exit code is 0 then check runasadmin or restart
-                            if (IsRunAsAdmin())
+                            //in case of crash try to restart the program
+                            if(niceHashMiner.ExitCode != 0)
```
Comment (nathsmaR) [inline src/NiceHashMinerLauncher/App.xaml.cs]: 3N4FAhGGXn8Q39kWhBQeNKQcEfM2NiqP4C

File: src/NiceHashMinerLauncher/App.xaml.cs (hunk 1 above)
Comment (tabotabo8) [reply src/NiceHashMinerLauncher/App.xaml.cs]: 365L1ZToRqrtJaDANjLXAQB7i5M2Us3ZNn

File: src/NiceHashMinerLauncher/App.xaml.cs (hunk 2 above)
Comment (tabotabo8) [reply src/NiceHashMinerLauncher/App.xaml.cs]: 365L1ZToRqrtJaDANjLXAQB7i5M2Us3ZNn

**Pull request merged**